src/NonPhpFile/NonPhpFileProcessor.php

```diff
@@ -4,40 +4,27 @@
 
 namespace Rector\Core\NonPhpFile;
 
-use Rector\Core\Configuration\RenamedClassesDataCollector;
 use Rector\Core\Contract\Processor\FileProcessorInterface;
+use Rector\Core\Contract\Rector\NonPhpRectorInterface;
 use Rector\Core\ValueObject\Application\File;
 use Rector\Core\ValueObject\StaticNonPhpFileSuffixes;
-use Rector\PSR4\Collector\RenamedClassesCollector;
 
 /**
  * @see \Rector\Tests\Renaming\Rector\Name\RenameClassRector\RenameNonPhpTest
  */
 final class NonPhpFileProcessor implements FileProcessorInterface
 {
     /**
-     * @var RenamedClassesDataCollector
+     * @var NonPhpRectorInterface[]
      */
-    private $renamedClassesDataCollector;
+    private $nonPhpRectors = [];
 
     /**
-     * @var RenamedClassesCollector
+     * @param NonPhpRectorInterface[] $nonPhpRectors
      */
-    private $renamedClassesCollector;
-
-    /**
-     * @var NonPhpFileClassRenamer
-     */
-    private $nonPhpFileClassRenamer;
-
-    public function __construct(
-        RenamedClassesDataCollector $renamedClassesDataCollector,
-        RenamedClassesCollector $renamedClassesCollector,
-        NonPhpFileClassRenamer $nonPhpFileClassRenamer
-    ) {
-        $this->renamedClassesDataCollector = $renamedClassesDataCollector;
-        $this->renamedClassesCollector = $renamedClassesCollector;
-        $this->nonPhpFileClassRenamer = $nonPhpFileClassRenamer;
+    public function __construct(array $nonPhpRectors)
+    {
+        $this->nonPhpRectors = $nonPhpRectors;
     }
```

Comment:
@lulco @sabbelasichon Hey guys, this it the contract refactoring between file processor and Rectors I was talking about.

Before it was impossible to extended `NonPhpFileProcessor` without modifing it here (SOLID violation). 
Now we just add a new service that implements `NonPhpRectorInterface` and that's it :+1: